File: prototype/Assets/microcosmicWar/Scripts/zz/GUI/zzWindow.js


class zzWindow extends zzGUIContainer
{

	//var windowRect = Rect (20, 20, 120, 50);
	var title : GUIContent;
	var useDefaultStyle=false;
	var  style : GUIStyle;
	var ID=0;
	
	var enableDrag=false;

	//var subElements:zzGUI[];

	
	virtual function impGUI():void
	{
		if(useDefaultStyle)
		{
			//print("useDefaultStyle");
			position = GUI.Window (ID, getPosition(), DoMyWindow,title);
			return;
		}
			//print("not useDefaultStyle");
		position = GUI.Window (ID, getPosition(), DoMyWindow,title,style);
	}
	
	virtual function DoMyWindow (windowID : int)
	{
		impSubs();
			//print("enableDrag:"+enableDrag);
		if(enableDrag)
			GUI.DragWindow ();
	}
	

	// Make the contents of the window
	//function DoMyWindow (windowID : int) 
//	{
	//if (GUI.Button (Rect (10,20,100,20), "Hello World"))
	//print ("Got a click");
		//for(var i:zzGUI in subElements)
		//	i.impGUI();
			//print("DoMyWindow begin");
			
			/*
		for(var i:Transform in transform)
		{
			//print(i.name);
			var impGUI:zzGUI = i.GetComponent(zzGUI);
			if(impGUI)
				impGUI.impGUI();
		}
		*/
			//print("DoMyWindow end");
//	}

}

File: prototype/Assets/microcosmicWar/Scripts/zz/GUI/zzWindow.cs

using UnityEngine;
using System.Collections;

public class zzWindow : zzGUIContainer
{

    public zzGUIStyle ContentAndStyle = new zzGUIStyle();
    public int ID = 0;

    public bool enableDrag = false;

    public bool alwayFront = false;

    /// <summary>
    /// 父UI有Group 时,将返回错误的信息
    /// </summary>
    public override bool isCursorOver
    {
        get 
        {
            var lMousePosition = Input.mousePosition;
            lMousePosition.y = Screen.height - lMousePosition.y;
            return position.Contains(lMousePosition); 
        }
    }

    public override Vector2 originOfCoordForSub
    {
        get
        {
            var lScreenPosition = screenPosition;
            return new Vector2(lScreenPosition.x, lScreenPosition.y);
        }
    }


    public override void impGUI(Rect rect)
    {
        if (ContentAndStyle.UseDefaultStyle)
        {
            position = GUI.Window(GetInstanceID(), rect, DoMyWindow, ContentAndStyle.Content);
            return;
        }
        //print("not _useDefaultStyle");
        position = GUI.Window(GetInstanceID(), rect, DoMyWindow, ContentAndStyle.Content, ContentAndStyle.Style);
    }

    public void DoMyWindow(int windowID)
    {
        impWindow(windowID);
        //print("enableDrag:"+enableDrag);
        if (enableDrag)
            GUI.DragWindow();
        if (alwayFront)
            GUI.BringWindowToFront(GetInstanceID());
    }

    
    public virtual void impWindow(int windowID)
    {
        impSubs();
    }

}